torchinductor/ir.py

```diff
@@ -1060,6 +1123,118 @@ def decide_layout(self):
         if isinstance(self.layout, FlexibleLayout):
             self.freeze_layout()
 
+    def simplify_loops(self):
+        _, (index_vars, reduce_vars) = dependencies.index_vars(
+            self.data.get_size(), self.data.get_reduction_size()
+        )
+        read_writes = self.get_read_writes()
+        index_formulas = list(
+            {
+                dep.index
+                for dep in itertools.chain(
+                    read_writes.reads,
+                    read_writes.writes,
+                    read_writes.index_exprs,
+                )
+            }
+        )
+        memory_addrs = [
+            dep.index for dep in itertools.chain(read_writes.reads, read_writes.writes)
+        ]
+
+        def simplify_and_reorder(vars_, sizes):
```

Comment:
Do we have a flag to control this? At least for the reordering part? I am thinking about future auto-tuning.

Reply:
`torchinductor.config.pick_loop_orders` turns on/off the reordering.

In the future we should definitely autotune the actual orders, or use some sort of global solver-based system.